apps/emqx_rule_engine/src/emqx_rule_engine_api.erl

```diff
@@ -423,7 +423,22 @@ param_path_id() ->
 %% Internal functions
 %%------------------------------------------------------------------------------
 
-err_msg(Msg) -> emqx_misc:readable_error_msg(Msg).
+err_msg({RuleError, {_E, Reason, _S}}) ->
```

Comment:
We _could_ move this functionality to `emqx_misc:readable_error_msg` because it seems to be only used from API modules. Unfortunately this is not obvious and also it would kinda leaking abstractions then.